### PR TITLE
feat(x-adapter): Add `sendParamsInBody` option to `RequestOptions`

### PR DESCRIPTION
**Pull request by tajespasarela**
EX-6450

Add sendParamsInBody to the x-adapter RequestOptions

The idea is to send parameters in the body instead in the url of the request if this new sendParamsInBody option is true

### CODE REVIEW

File: packages/x-adapter/src/http-clients/__tests__/fetch.http-client.spec.ts [hunk 1]
```diff
@@ -106,16 +107,33 @@ describe('fetch httpClient testing', () => {
...
 /**
  * Expects the `fetch` function to be called with the passed `URL`.
  *
  * @param url - The `URL` to check.
+ * @param options - The `options` to check.
  *
  * @internal
  */
-function expectFetchCallWith(url: string): void {
+function expectFetchCallWith(url: string, options = {}): void {
```
Comment (javieri-empathy) [inline packages/x-adapter/src/http-clients/__tests__/fetch.http-client.spec.ts:136]: Type the parameter
```suggestion
function expectFetchCallWith(url: string, options: RequestInit = {}): void {
```

File: packages/x-adapter/src/http-clients/__tests__/fetch.http-client.spec.ts [hunk 2]
```diff
@@ -106,16 +107,33 @@ describe('fetch httpClient testing', () => {
...
 });
 
 /**
  * Expects the `fetch` function to be called with the passed `URL`.
  *
  * @param url - The `URL` to check.
+ * @param options - The `options` to check.
  *
  * @internal
  */
```
Comment (javieri-empathy) [inline packages/x-adapter/src/http-clients/__tests__/fetch.http-client.spec.ts:135]:
```ts
/**
 * Expects the `fetch` function to be called with the passed `URL`.
 *
 * @param url - The `URL` to check.
 * @param options - An optional and partial `fetch` request init options to assert that fetch has been called with those parameters.
 *
 * @internal
 */
```

**Review by javieri-empathy — CHANGES_REQUESTED** [review]

File: packages/x-adapter/src/http-clients/__tests__/fetch.http-client.spec.ts (hunk 2 above)
Comment (tajespasarela) [reply packages/x-adapter/src/http-clients/__tests__/fetch.http-client.spec.ts]: done

File: packages/x-adapter/src/http-clients/__tests__/fetch.http-client.spec.ts (hunk 1 above)
Comment (tajespasarela) [reply packages/x-adapter/src/http-clients/__tests__/fetch.http-client.spec.ts]: done

File: packages/x-adapter/src/http-clients/__tests__/fetch.http-client.spec.ts [hunk 3]
```diff
@@ -106,16 +107,33 @@ describe('fetch httpClient testing', () => {
...
-function expectFetchCallWith(url: string): void {
+function expectFetchCallWith(url: string, options = {}): void {
   expect(window.fetch as jest.Mock).toHaveBeenCalledTimes(1);
-  expect(window.fetch as jest.Mock).toHaveBeenCalledWith(url, expect.anything());
+  expect(window.fetch as jest.Mock).toHaveBeenCalledWith(url, expect.objectContaining(options));
```
[the remark this answers is not in the document]
Comment (tajespasarela) [reply packages/x-adapter/src/http-clients/__tests__/fetch.http-client.spec.ts]: changed

File: packages/x-adapter-platform/src/mappers/requests/tagging-request.mapper.ts [hunk 4]
```diff
@@ -1,7 +1,7 @@
-import { schemaMapperFactory } from '@empathyco/x-adapter';
+import { Mapper } from '@empathyco/x-adapter';
 import { TaggingRequest } from '@empathyco/x-types';
-import { taggingRequestMutableSchema } from '../../schemas/requests/tagging-request.schema';
 
-export const taggingRequestMapper = schemaMapperFactory<TaggingRequest, any>(
-  taggingRequestMutableSchema
-);
+export const taggingRequestMapper: Mapper<TaggingRequest, any> = ({
```
Comment (javieri-empathy) [inline packages/x-adapter-platform/src/mappers/requests/tagging-request.mapper.ts:4]: Is the wrapping of the params object intentional?

Also, due to this change, the `tagging-request-mapper.spec.ts` test fails:

```ts
 it('should map the request', () => {
    const internalRequest: TaggingRequest = {
      url: 'https://api.staging.empathy.co/tagging/v1/track/empathy/click',
      params: {
        filtered: 'false',
        follow: false,
        lang: 'en',
        origin: 'search_box:none',
        page: '1',
        position: '1',
        productId: '12345-U',
        q: '12345',
        scope: 'desktop',
        spellcheck: 'false',
        title: 'Xoxo Women Maroon Pure Georgette Solid Ready-to-wear Saree'
      }
    };

    expect(taggingRequestMapper(internalRequest, {})).toStrictEqual({
      params: {
        filtered: 'false',
        follow: false,
        lang: 'en',
        origin: 'search_box:none',
        page: '1',
        position: '1',
        productId: '12345-U',
        q: '12345',
        scope: 'desktop',
        spellcheck: 'false',
        title: 'Xoxo Women Maroon Pure Georgette Solid Ready-to-wear Saree'
      }
    });
```

**Review by javieri-empathy — CHANGES_REQUESTED** [review]

File: packages/x-adapter-platform/src/mappers/requests/tagging-request.mapper.ts (hunk 4 above)
Comment (tajespasarela) [reply packages/x-adapter-platform/src/mappers/requests/tagging-request.mapper.ts]: no, it was not intended, my fault.

Fixed now

**Review by javieri-empathy — APPROVED** [review]

**Pull request merged**